common/v2/features/CreateWallet/Mnemonic/components/GeneratePhrasePanel.tsx

```diff
@@ -1,50 +1,139 @@
 import React, { Component } from 'react';
 import { Button, Typography } from '@mycrypto/ui';
+import styled from 'styled-components';
 
-import { ContentPanel } from 'v2/components';
-import './GeneratePhrasePanel.scss';
+import { ExtendedContentPanel } from 'v2/components';
+import { PanelProps } from '../../CreateWallet';
+import translate, { translateRaw } from 'translations';
 
 // Legacy
 import reloadIcon from 'common/assets/images/icn-reload.svg';
-import { MnemonicStageProps } from '../constants';
+import Spinner from 'components/ui/Spinner';
+
+const DescriptionItem = styled(Typography)`
+  margin-top: 18px;
+  font-weight: normal;
+  font-size: 18px !important;
+
+  strong {
+    font-weight: 900;
+  }
+`;
+
+const RegenerateImage = styled.img`
+  width: 16px;
+  height: 16px;
+  margin-right: 10px;
+  pointer-events: none;
+  display: inline;
+`;
+
+const GeneratePhrasePanelWords = styled(Typography)`
+  padding: 29px;
+  border: 1px solid #e5ecf3;
+  line-height: 1.5;
+  text-align: center;
+  font-weight: 500;
+  word-spacing: 14px;
+  height: 140px;
+  display: flex;
+  align-items: center;
+`;
+
+const Label = styled.p`
+  font-size: 20px;
+  font-weight: 500;
+  margin-top: 42px;
+`;
+
+const ButtonsWrapper = styled.div`
+  margin-top: 32px;
+  display: flex;
+  flex-direction: column;
+`;
+
+const StyledButton = styled(Button)`
+  font-size: 18px;
+  width: 100%;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  margin-top: 15px;
+  &:focus,
+  &:hover {
+    img {
+      filter: brightness(0) invert(1);
+    }
+  }
+`;
+
+const ButtonWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  width: 100%;
+  height: 66px;
+`;
+
+interface Props extends PanelProps {
+  words: string[];
+  generateWords(): void;
+  decryptMnemonic(): Promise<string>;
+}
+
+export default class GeneratePhrasePanel extends Component<Props> {
+  public state = { decrypting: false };
 
-export default class GeneratePhrasePanel extends Component<
-  { words: string[]; generateWords(): void } & MnemonicStageProps
-> {
   public componentDidMount() {
     const { generateWords } = this.props;
-
     generateWords();
   }
 
+  public handleNextClick = () => {
+    const { onNext, decryptMnemonic } = this.props;
+
+    this.setState({ decrypting: true });
+    setTimeout(() => {
+      decryptMnemonic();
```

Comment:
In the decryptMnemonic() we call the mnemonicToSeed() from the bip39 library. This is a blocking function which seems to prevent the loading indicator to be rendered, therefore timeout is used. Returning a promise in the decryptMnemonic() does not help either, another alternative would be a web worker or some CSS workarounds, both of which seem too complex solutions for this kind of problem.